### DISCUSSION TITLE
MoveCraft - 基于 Move NFT 的合成游戏

### DISCUSSION THREAD

**Discussion by jolestar** [NonceGeek/Move-Camp#167] [Amazing Things!]
MoveCraft 的目标是尝试探索一下 Move 上的 NFT 组合合成玩法

思路有两个：

1. 定义一种 NFT，NFT 由 MoveCraft 发行，但其中内置类型来区分不同的物品，然后实现合成。合成规则参照 Minecraft 里的机制，通过排列组合来定义合成规则。
2. 尝试支持任意 NFT 的合成，和链上其他 NFT 合作起来。

第一种方案实现简单一些，第二种稍微复杂一些。

有一个初步的样板，但尚未完成，源码参看 https://github.com/movecraftorg/movecraft ，感兴趣的朋友可以一起来完成。

**Comment by golden-bull** [comment]
毛遂自荐一下，之前做过几个NFT项目，使用Solidity实现过不同类别NFT的关联，以及NFT的升级、合成，不过感觉不是很模块化。
作为对合成玩法很感兴趣的MOVE新手，希望能加入这个队伍中。
前端、后端、合约都可以做，不过水平都一般，一切都听队长安排。

**Reply by leeduckgo** [reply]
👏👏

**Comment by pause125** [comment]
我也在想这么一个游戏，没想到 @jolestar 老早已经写了这么多了~

**Reply by jolestar** [reply]
没完成😅

**Comment by zhaoge5859** [comment]
新手可报名吗

**Reply by jolestar** [reply]
可以可以

**Comment by v1xingyue** [comment]
标记一下，有时间参与一下。 凑个热闹

**Comment by jolestar** [comment]
这两天有点忙，我周末再折腾一下看看。

**Comment by menghonghan** [comment]
同新手，请问可以报名吗！

**Reply by jolestar** [reply]
可以呀

**Comment by jolestar** [comment]
重构了一下，加了个测试用例，大体实现逻辑应该清楚了。初步计划先实现方案1版本，之后再考虑如何演化成方案2.

加了几个 issue https://github.com/movecraftorg/movecraft/issues ，感兴趣的朋友可以来参与。
另外会前端的朋友可以考虑构建个简单的前端。

**Reply by leeduckgo** [reply]
Great!

**Comment by jolestar** [comment]
创建了个 tg 群，感兴趣的朋友加一下，看看后面怎么协作 https://t.me/movecraft
